docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,112 @@
+
+Details of client interfaces to support these methods will likely vary by language. For example, the [Java client](java-client.md)
+uses [Reactive Streams](http://www.reactive-streams.org/) for its asynchronous streaming interface. 
+
+### Terminating push queries ###
+
+Requests to terminate push queries should be sent to the [`/close-query` server endpoint](TODO).
+The client should expose an interface that allows users to pass in a push query ID to be passed to this endpoint.
+
+### Inserting new rows of data into existing ksqlDB streams ###
+
+ksqlDB supports inserting new rows of data into existing ksqlDB streams via the [`/inserts-stream` server endpoint](TODO).
+If the server successfully validates the target stream name from a request made to the `/inserts-stream` endpoint,
+the client may then write JSON-serialized rows of data to the connection, and the server will insert each
+asynchronously and write an ack containing the relevant sequence number to the response.
+
+Because the insertions happen asynchronously on the server side, it makes sense for the client to expose APIs
+for inserting one row at a time, but inserting batches of rows is less useful due to the lack of transactional guarantees.
+
+The client should provide a convenient way for users to create rows of data to insert.
+The Java client's [KsqlObject](TODO) is one example, though details will likely vary by language.
+
+### Creation of new streams, tables, and persistent queries ###
+
+These DDL/DML requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+### Terminating persistent queries ###
+
+These requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+### Listing existing streams, tables, topics, and queries ###
+
+These admin operation requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+Configurations
+--------------
+
+The client of course needs to expose options for specifying the address of the ksqlDB server to connect to.
+Additional options important to expose include: 
+- Support for TLS-enabled ksqlDB servers
+- Support for mutual-TLS-enabled ksqlDB servers
+- Support for ksqlDB servers with HTTP basic authentication enabled
+
+It may also be nice to support configuring multiple ksqlDB server addresses, so the client may route requests
+to a different server if one is down.
+
+Process
+-------
+
+Thanks for your interest in contributing a client!
+
+To get started:
+- Open [KLIP](../../../design-proposals/README.md) to propose the client you'd like to implement. The KLIP should include a high-level design and example interfaces.
+- Contribute code to [ksql repo](https://github.com/confluentinc/ksql)
+- Testing: Besides unit tests in the relevant language, there should also be integration tests to spin up a ksqlDB server and validate client behavior. TODO how to make this easier?
```

Comment:
Hey @MichaelDrogalis we'll have to think about how to make the addition of integration tests in other languages easier. It's not clear to me how to do this right now. Is there an established way to run code from multiple languages (the ksqlDB server in Java + the client in a non-Java language) for integration tests?

Reply:
I'm really not sure. Might be a good thing to bring up at the next team meeting, I'm sure we can brainstorm ideas.